### PR TITLE
Add db form

### PR DESCRIPTION
**Pull request by seanrichardson95**
Wow, I hate merge conflict and rebasing...

In this pull request I setup a connection to the event tracking DB via the UI via our API. There are three new methods in _ApiClient.js_, `connectToDB`, `removeDBConnection`, and `checkDBConnection` that send POST, DELETE, and GET requests to the "api/connection" endpoint.

`checkDBConnection` is called on the first render of `App`, and we're currently connected to the DB, it'll save the name of the DB as state.

I created two new components, `DBModal` and `DatabaseConnection`. 

`DBModal` is a modal that pops up to submit DB credentials to our API. 
- It has a little bit of validation on it (can't leave any field empty and must have a proper hostname)
- As of right now if you send improper data, an error occurs on our backend and the backend goes down. Not sure why this happens as of right now. I think it has to be fixed on the backend.

`DatabaseConnection` is a wrapper for the "Connect to Database"/"Disconnect" button, that'll render one way or the other depending on if we're currently connected to the DB.
- I'm pretty tired at the moment so I'm not going to try and figure out how to get it to render on the right side of the page, but hopefully you can show me tomorrow @julia-martin 

As of right now I'm initializing the `dbName` state on the top level in `App`. Not sure if that's the best place for it.

I also made a small change to the `deleteFlag` button, so instead of an `alert()` it uses a `window.confirm()` so you can hit cancel if you change your mind about deleting a flag.

### CODE REVIEW

**Comment by seanrichardson95** [comment]
Also, I kept getting a 500 Error object when submitting the wrong credentials. I couldn't figure out how to pull the error message out of that, so that hasn't been implemented as of right now.

**Comment by julia-martin** [comment]
Great work! It's nice that you did the form validation on the frontend. Also good catch on the `alert` vs `confirm`, makes way more sense this way! I just found one typo, I'll add a comment on that line

File: client/src/components/DBModal.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          </div>
+          <div className="mt-2.5">
+            <label htmlFor="password" className="mr-5">Password: </label>
+            <input id="password" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={password} onChange={(e) => setPassword(e.target.value)} />
+          </div>
+          <div className="mt-2.5">
+            <label htmlFor="database" className="mr-5">Database: </label>
+            <input id="database" type="text" className="border border-primary-oxfordblue rounded-lg px-2" value={database} onChange={(e) => setDatabase(e.target.value)} />
+          </div>
+          <div className="mt-2.5" className="mr-5">
```
Comment (julia-martin) [inline client/src/components/DBModal.jsx]: This div on line 75 has `className` twice, which is breaking the CSS for this div. I think you meant to put the second className on the `label` element

**Review by julia-martin — CHANGES_REQUESTED** [review]

File: client/src/components/DBModal.jsx (hunk 1 above)
Comment (seanrichardson95) [reply client/src/components/DBModal.jsx]: Good catch! Just fixed it.

**Review by julia-martin — APPROVED** [review]

**Comment by julia-martin** [comment]
You don't have to do this now obviously, but to answer your question about the positioning, you can just add `flex justify-between items-center` to the header's class, turning it into a Flex container
css

**Review by calebrs — APPROVED** [review]
Nice! Everything looks good to me!

**Pull request merged**